components/VM_Arm/src/main.c

```diff
@@ -826,6 +854,17 @@ static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, c
             paths = camkes_dtb_get_node_paths(&num_paths);
         }
 
+#ifdef CONFIG_VM_CUSTOM_DTB_BASE
+        err = get_file(linux_image_config.dtb_base, linux_gen_dtb_base_buf);
+        if (!err) {
```

Comment:
It feels semantically wrong/misleading to use the expression `!err` here. Might be better to use 
```
int dtb_len = get_file(...);
if (0 == dtb_len) { ... error 
```